### DISCUSSION TITLE
虽然是目录程序，作者能增加分享直连有效时长和访问密码吗？

### DISCUSSION THREAD

**Discussion by Contribuv** [AlistGo/alist#2524] [Ideas]
这个功能很重要，因为 有的文件 未对 访客公开，但是有时候需要分享出去，目前是没办法设置分享时长和分享密码。
现在登录后分享出去的直连，访客也能正常下载，很容易被爬虫。

**Comment by wscqstu** [comment]
密码是可以的，通过元信息来设置密码，时长我不知道怎么设置。

**Reply by Contribuv** [reply]
你说的是对访客开放的目录设置可以通过元信息设置文件夹密码，我所说的是 未对访客开放的某个文件 分享出去 设置有效时长和密码，我提这个功能是因为现在可以通过访客权限直接把挂载盘所有内容 给 爬虫 下来，非常不安全

**Reply by shaojs321** [reply]
可以通过服务器自己生成短链接，并通过控制短链接的有效时长和密码来解决这个问题，自己生成短链接网上有开源方案
